pytest_bdd/plugin.py

```diff
@@ -47,14 +49,19 @@ def add_bdd_ini(parser):
 @pytest.mark.trylast
 def pytest_configure(config):
     """Configure all subplugins."""
+    ConfigStack.add(config)
```

Comment:
It is all here to fix tests that we run using `testdir` plugin.

Our problem is that:
1. When we start all tests it calls `pytest_configure`.
2. Then when it starts some test with `testdir` it calls `pytest_configure` again with new config.
3. After test (as in current code there is no `pytest_unconfigure`) it just runs another test.
4. As `pytest_configure` from point 2 changed pytest config - our tests start to fail as they expect default config (from point 1).

I added `pytest_unconfigure` hook so now after running test using `testdir` it can bring back previously used config.

So this pull request changes logic to:
1. When we start all tests it calls `pytest_configure` (we store config on stack).
2. Then when it starts some test with `testdir` it calls `pytest_configure` again with new config (again we store new config on stack).
3. After tests it calls `pytest_unconfigure` -> we bring back previous config from the stack.
4. We have previous config back and tests pass :)